app/streaming/video/ffmpeg-renderers/eglvid.cpp

```diff
@@ -0,0 +1,480 @@
+ *  - https://www.khronos.org/registry/OpenGL/extensions/OES/OES_EGL_image_external.txt
+ *  - https://gist.github.com/rexguo/6696123
+ *  - https://wiki.libsdl.org/CategoryVideo
+ */
+
+static QStringList egl_get_extensions(EGLDisplay dpy) {
+    const auto egl_extensions_str = eglQueryString(dpy, EGL_EXTENSIONS);
+    if (!egl_extensions_str) {
+        SDL_LogWarn(SDL_LOG_CATEGORY_APPLICATION, "Unable to get EGL extensions");
+        return QStringList();
+    }
+    return QString(egl_extensions_str).split(" ");
+}
+
+EGLExtensions::EGLExtensions(EGLDisplay dpy) :
+    m_extensions(egl_get_extensions(dpy))
+{}
+
+bool EGLExtensions::is_supported(const QString &extension) const {
+    return m_extensions.contains(extension);
+}
+
+EGLRenderer::EGLRenderer(IFFmpegRenderer *frontend_renderer)
+    :
+        m_SwPixelFormat(AV_PIX_FMT_NONE),
+        m_egl_display(nullptr),
+        m_textures{0},
+        m_shader_program(0),
+        m_context(0),
+        m_window(nullptr),
+        m_frontend(frontend_renderer),
+        m_vao(0),
+        m_colorspace(AVCOL_SPC_NB),
+        m_color_full(false),
+        EGLImageTargetTexture2DOES(nullptr)
+{
+    SDL_assert(frontend_renderer);
+    SDL_assert(frontend_renderer->canExportEGL());
+}
+
+EGLRenderer::~EGLRenderer()
+{
+    deinitialize();
+}
+
+bool EGLRenderer::prepareDecoderContext(AVCodecContext*, AVDictionary**)
+{
+    /* Nothing to do */
+
+    SDL_LogInfo(SDL_LOG_CATEGORY_APPLICATION,
+                "Using EGL renderer!!");
+
+    return true;
+}
+
+void EGLRenderer::notifyOverlayUpdated(Overlay::OverlayType)
+{
+    // TODO: FIXME
+}
+
+bool EGLRenderer::isRenderThreadSupported()
+{
+    // TODO: FIXME (maybe ?)
+    return false;
+}
+
+bool EGLRenderer::isPixelFormatSupported(int, AVPixelFormat pixelFormat)
+{
+    // Remember to keep this in sync with EGLRenderer::renderFrame()!
+    switch (pixelFormat)
+    {
+    case AV_PIX_FMT_NV12:
+        return true;
+    default:
+        return false;
+    }
+}
+
+static GLuint load_and_build_shader(GLenum shader_type,
+                    const char *file) {
+    GLuint shader = glCreateShader(shader_type);
+    if (!shader || shader == GL_INVALID_ENUM)
+        return 0;
+
+    auto source_data = Path::readDataFile(file);
+    GLint len = source_data.size();
+    const char *buf = source_data.data();
+
+    glShaderSource(shader, 1, &buf, &len);
+    glCompileShader(shader);
+    GLint status;
+    glGetShaderiv(shader, GL_COMPILE_STATUS, &status);
+    if (!status) {
+        char shader_log[512];
+        glGetShaderInfoLog(shader, sizeof (shader_log), nullptr, shader_log);
+        SDL_LogError(SDL_LOG_CATEGORY_APPLICATION,
+                 "EGLRenderer: cannot load shader \"%s\": %s",
+                 file, shader_log);
+        return 0;
+    }
+
+    return shader;
+}
+
+bool EGLRenderer::compileShader() {
+    if (m_shader_program) {
+        glDeleteProgram(m_shader_program);
```

Comment:
Is there a code path where we can actually reach this case?

Reply:
Can the read-back pixel format change between two `renderFrame` call ?